File: PostBox/src/main/java/com/mvp/view/Window.java
package com.mvp.view;

import com.mvp.presenter.BoxPresenter;

public class Window implements IBoxView {

    private BoxPresenter boxPresenter;

    public Window() {
        this.boxPresenter = new BoxPresenter(this);
    }

    protected void receivedBox(float length, float width, float height) {
        System.out.println("View : received a box");
        boxPresenter.findSuitableBoxFromModel(length, width, height);
    }

    @Override
    public void reportResultToClient(String box) {
        System.out.println("View : show the box => " + box);
    }
}

File: PostBox/src/main/kotlin/com/mvp/view/Window.kt
package com.mvp.view.kt

import com.mvp.presenter.kt.BoxPresenter

class Window : IBoxView {

    private val boxPresenter: BoxPresenter? by lazy { BoxPresenter(this) }

    fun receivedBox(length: Float?, width: Float?, height: Float?) {
        println("View : received a box")
        boxPresenter?.findSuitableBoxFromModel(length, width, height)
    }

    override fun reportResultToClient(result: String) {
        println("View : show the box => $result")
    }
}